hibernate-core/src/main/java/org/hibernate/metamodel/internal/EmbeddableRepresentationStrategyPojo.java

```diff
@@ -36,37 +42,65 @@
 import org.hibernate.type.internal.CompositeUserTypeJavaTypeWrapper;
 import org.hibernate.usertype.CompositeUserType;
 
+import static org.hibernate.internal.util.NullnessUtil.castNonNull;
+
 /**
  * @author Steve Ebersole
  */
-public class EmbeddableRepresentationStrategyPojo extends AbstractEmbeddableRepresentationStrategy {
-	private final StrategySelector strategySelector;
+public class EmbeddableRepresentationStrategyPojo implements EmbeddableRepresentationStrategy {
+	private final JavaType<?> embeddableJavaType;
+	private final PropertyAccess[] propertyAccesses;
+	private final Map<String,Integer> attributeNameToPositionMap;
 
+	private final StrategySelector strategySelector;
 	private final ReflectionOptimizer reflectionOptimizer;
-	private final EmbeddableInstantiator instantiator;
+	private final Map<String, EmbeddableInstantiator> instantiators;
 
 	public EmbeddableRepresentationStrategyPojo(
 			Component bootDescriptor,
 			Supplier<EmbeddableMappingType> runtimeDescriptorAccess,
 			EmbeddableInstantiator customInstantiator,
 			CompositeUserType<Object> compositeUserType,
 			RuntimeModelCreationContext creationContext) {
-		super(
-				bootDescriptor,
-				resolveEmbeddableJavaType( bootDescriptor, compositeUserType, creationContext ),
-				creationContext
-		);
+		this.embeddableJavaType = resolveEmbeddableJavaType( bootDescriptor, compositeUserType, creationContext );
 
+		final int propertySpan = bootDescriptor.getPropertySpan();
+		this.propertyAccesses = new PropertyAccess[propertySpan];
+		this.attributeNameToPositionMap = new ConcurrentHashMap<>( propertySpan );
```

Comment:
I think a regular hash map should be fine

```suggestion
		this.attributeNameToPositionMap = new HashMap<>( propertySpan );
```

Reply:
This was copied from `AbstractEmbeddableRepresentationStrategy`, which I merged into EmbeddableRepresentationStrategyPojo` since it was the only implementor and it simplified the code. I will try a normal `HashMap` if you think it's better.